app/src/main/java/com/pileproject/drive/execution/ExecutionCondition.java

```diff
@@ -17,28 +17,19 @@
 package com.pileproject.drive.execution;
 
 
-import android.content.Context;
-
-import com.pileproject.drive.R;
-import com.pileproject.drive.app.DriveApplication;
 import com.pileproject.drive.programming.visual.block.BlockBase;
 
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Stack;
 
 /**
- * This class has the condition of execution
- *
- * @author <a href="mailto:tatsuyaw0c@gmail.com">Tatsuya Iwanari</a>
- * @version 1.0 7-July-2013
+ * A container class that has the condition of program execution.
  */
 public class ExecutionCondition {
```

Comment:
I'm not sure what kind of use cases you expect,
but I do not like these access controls since it does not protect itself.
And this seems provide inconsistent interfaces; while ifStack is public, you also add a kind of setter for that.
In short, why not encapsulate this?